v2/protocol/http/protocol.go

```diff
@@ -218,6 +223,7 @@ type msgErr struct {
 func (p *Protocol) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	m := NewMessageFromHttpRequest(req)
 	if m == nil || m.ReadEncoding() == binding.EncodingUnknown {
+		http.Error(rw, "Cannot read CloudEvent: "+binding.ErrUnknownEncoding.Error(), http.StatusUnsupportedMediaType)
```

Comment:
ok sorry, lets hold off on this change until we decide on a fix in #483 

There is value in letting the client level know that an event was attempted and it failed... the user or client could choose to fix the event for the consumer. So let's hold off on http.Error for now. Sorry for the flip flopping.... The reason is just because we are trying to not put too many runtime choices at the protocol level.

Reply:
Removed